### PR TITLE
fix(swc): #2211

### PR DESCRIPTION
**Pull request by powerfulyang**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #2211 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: lib/compiler/swc/swc-compiler.ts [hunk 1]
```diff
@@ -170,6 +171,21 @@ export class SwcCompiler extends BaseCompiler {
       swcOptions.jsc.baseUrl = path.join(rootDir, swcOptions.jsc.baseUrl);
     }
 
+    if (swcOptions?.jsc?.paths) {
```
Comment (kamilmysliwiec) [inline lib/compiler/swc/swc-compiler.ts:174]: This condition will always be true for someone who has paths defined in the tsconfig.json file (which is required for DX)

Comment (powerfulyang) [reply lib/compiler/swc/swc-compiler.ts:174]: Yes, `swcRcFile?.jsc?.paths` will throw error when paths didn't defined in `.swcrc`. This is the `@swc/cli` BUG in Windows, if pass an object to `swcCli.default` function.

**Comment by powerfulyang** [comment]
This [commit](https://github.com/nestjs/nest-cli/commit/0cc6c929e956740fad57443a5622f908c8d7a178) didn't fix it.
If `swcOptions` is an object, even if `paths` are defined in the `.swcrc` file, it does not function properly; it only works when `swcOptions` is set to `undefined`.

**Comment by powerfulyang** [comment]
> https://github.com/swc-project/swc/issues/7806
The issue may have been resolved; further testing will be conducted to confirm.

**Comment by powerfulyang** [comment]
This issue appears to be a bug in `@swc/core@1.3.80`, and it does not support the `paths` feature as expected. As a workaround, setting `swcOptions` to `undefined` serves as a temporary fix. Unfortunately, this is not something that can be resolved on the SWC side until the bug is addressed in a future release.

**Comment by viniciussantos45** [comment]
After updating `@swc/core` to version `1.3.81` it started working for `@nestjs/cli` version `10.1.16`
## My current setup:
### tsconfig.json
```json
{
  "compilerOptions": {
   ...
    "outDir": "./dist",
    "baseUrl": ".",
    "paths": {
      "@common/*": ["src/common/*"],
      "@config/*": ["src/config/*"],
      "@modules/*": ["src/modules/*"]
    }
   ...
  }
}

```

### .swcrc
```json
{
  "$schema": "https://json.schemastore.org/swcrc",
  "sourceMaps": true,
  
  "minify": false,
  "jsc": {
    ...
    "baseUrl": "./",
    "paths": {
      "@common/*": ["src/common/*"],
      "@config/*": ["src/config/*"],
      "@modules/*": ["src/modules/*"]
    },
    ...
  }
}
```

### nest-cli-.json
```json
{
  "$schema": "https://json.schemastore.org/nest-cli",
  "collection": "@nestjs/schematics",
  "sourceRoot": "src",
  "compilerOptions": {
    "deleteOutDir": true,
    "builder": "swc",
    "typeCheck": true
  }
}
```

**Pull request closed without merging**